elixir/apps/api/lib/api/client/channel.ex

```diff
@@ -58,10 +49,21 @@ defmodule API.Client.Channel do
     OpenTelemetry.Tracer.set_current_span(opentelemetry_span_ctx)
 
     OpenTelemetry.Tracer.with_span "client.after_join" do
-      :ok = API.Endpoint.subscribe("client:#{socket.assigns.client.id}")
       :ok = Clients.connect_client(socket.assigns.client)
 
-      {:ok, resources} = Domain.Resources.list_authorized_resources(socket.assigns.subject)
+      {:ok, resources} = Resources.list_authorized_resources(socket.assigns.subject)
+
+      # We subscribe for all resource events but only care about update events,
+      # where resource might be renamed which should be propagated to the UI.
+      :ok = Enum.each(resources, &Resources.subscribe_for_events_for_resource/1)
```

Comment:
(nit: naming): In my head "subscribe to" seems more natural than "subscribe for". But don't feel strongly

```suggestion
      :ok = Enum.each(resources, &Resources.subscribe_to_events_for_resource/1)
```